white_balancer/src/white_balancer.cpp

```diff
@@ -0,0 +1,200 @@
+                                                    { 9400,  { 209, 223, 255 }},
+                                                    { 9500,  { 208, 222, 255 }},
+                                                    { 9600,  { 207, 221, 255 }},
+                                                    { 9700,  { 207, 221, 255 }},
+                                                    { 9800,  { 206, 220, 255 }},
+                                                    { 9900,  { 205, 220, 255 }},
+                                                    { 10000, { 207, 218, 255 }},
+                                                    { 10100, { 207, 218, 255 }},
+                                                    { 10200, { 206, 217, 255 }},
+                                                    { 10300, { 205, 217, 255 }},
+                                                    { 10400, { 204, 216, 255 }},
+                                                    { 10500, { 204, 216, 255 }},
+                                                    { 10600, { 203, 215, 255 }},
+                                                    { 10700, { 202, 215, 255 }},
+                                                    { 10800, { 202, 214, 255 }},
+                                                    { 10900, { 201, 214, 255 }},
+                                                    { 11000, { 200, 213, 255 }},
+                                                    { 11100, { 200, 213, 255 }},
+                                                    { 11200, { 199, 212, 255 }},
+                                                    { 11300, { 198, 212, 255 }},
+                                                    { 11400, { 198, 212, 255 }},
+                                                    { 11500, { 197, 211, 255 }},
+                                                    { 11600, { 197, 211, 255 }},
+                                                    { 11700, { 197, 210, 255 }},
+                                                    { 11800, { 196, 210, 255 }},
+                                                    { 11900, { 195, 210, 255 }},
+                                                    { 12000, { 195, 209, 255 }}};
+    // Dummy color
+    int temp = 1000;
+    // Image calback
+    void imageCallback(const sensor_msgs::ImageConstPtr& msg);
+    void set_temp(int temp);
+};
+
+WhiteBalancer::WhiteBalancer()
+{
+    // Dynamic reconfigure stuff
+    dynamic_reconfigure::Server<white_balancer::WhiteBalanceConfig> server;
+    dynamic_reconfigure::Server<white_balancer::WhiteBalanceConfig>::CallbackType f;
+    f = boost::bind(&WhiteBalancer::callbackRC, this, _1, _2);
+    server.setCallback(f);
+
+    // Register image messages
+    ros::NodeHandle nh;
+    image_transport::ImageTransport it(nh);
+
+    WhiteBalancer::pub = it.advertise("/image", 1);
+    image_transport::Subscriber sub = it.subscribe("/image_unbalanced", 1, &WhiteBalancer::imageCallback, this);
```

Comment:
Why are the topics not configurable? I don't care that much about the dynamic stuff, but it should be at least a YAML parameter. We do not want to recompile the whole thing for a different topic name!